src/sandbox/components/screens/call-functions/components/available-functions.tsx

```diff
@@ -0,0 +1,275 @@
+import { useContractInterface } from '@sandbox/components/screens/call-functions/components/use-contract-interface';
+
+const ArgLine = ({ name, type, handleChange, placeholder = name, ...rest }: any) => (
+  <Box width="100%" {...rest}>
+    <Input
+      width="100%"
+      type={getTypeString(type).includes('int') ? 'number' : 'text'}
+      name={name}
+      id={name}
+      onChange={handleChange}
+      placeholder={`${getTypeString(type)}`}
+    />
+  </Box>
+);
+
+const Label: React.FC<BoxProps> = props => (
+  <Text
+    as="label"
+    fontSize="12px"
+    fontWeight="500"
+    display="block"
+    color={color('text-caption')}
+    {...props}
+  />
+);
+
+const FunctionLine = ({ name, type, handleChange, ...rest }) => {
+  const tuple = type?.tuple || type?.optional?.tuple;
+  if (tuple) {
+    return (
+      <Box {...rest}>
+        <Stack id={name} isInline spacing="base" width="100%">
+          {tuple.map(arg => (
+            <Box flexGrow={1}>
+              <Label htmlFor={`${name}`} mb="tight">
+                ({name}): {arg.name}
+                {type?.optional ? ' (optional)' : ''}
+              </Label>
+              <ArgLine handleChange={handleChange} {...arg} name={`${name}.${arg.name}`} />
+            </Box>
+          ))}
+        </Stack>
+      </Box>
+    );
+  }
+  return (
+    <Box>
+      <Label mb="tight" htmlFor={name}>
+        {name}
+      </Label>
+      <ArgLine name={name} type={type} handleChange={handleChange} {...rest} />
+    </Box>
+  );
+};
+
+const Function = ({ func }) => {
+  const setFunctionName = useSetRecoilState(currentFunctionState);
+
+  return (
+    <>
+      <Flex
+        onClick={() => setFunctionName(func.name)}
+        justifyContent="space-between"
+        p="base"
+        borderBottom={border()}
+        _hover={{
+          cursor: 'pointer',
+          bg: color('bg-alt'),
+        }}
+      >
+        <Flex alignItems="center">
+          <Flex alignItems="center">
+            <Grid
+              placeItems="center"
+              border={border()}
+              borderRadius="100%"
+              size="32px"
+              color={color('text-caption')}
+            >
+              {func.access === 'read_only' ? <ApiIcon size="20px" /> : <FunctionIcon size="20px" />}
+            </Grid>
+            <Text
+              fontSize="14px"
+              fontFamily={`"Fira Code", monospace`}
+              ml="base"
+              fontWeight="500"
+              color={color('text-body')}
+            >
+              {func.name}
+            </Text>
+          </Flex>
+          {func.access === 'read_only' && (
+            <Badge ml="base" bg={color('bg')} border={border()} color={color('text-caption')}>
+              {func.access}
+            </Badge>
+          )}
+        </Flex>
+        <IconButton dark icon={ArrowRightIcon} />
+      </Flex>
+    </>
+  );
+};
+
+const ReadOnly = () => {
```

Comment:
~~Unrelated to this: if I am already in the "call function of existing contract" state and I try and "Connect stacks wallet", I auth and then nothing happens. I have to go back to the root state and then sign in.~~

Edit: cannot reproduce 🤷🏼

Reply:
If I am in the "call function" state, and I click the hamburger to see my account info, the 'call function' panel shrinks - seemingly unnecessarily? I'm honestly not sure what the "third panel" is there for

![image](https://user-images.githubusercontent.com/1109058/108551624-0f7db900-72a5-11eb-912a-20723e19bdab.png)